packages/gatsby-image/src/index.js

```diff
@@ -479,6 +484,11 @@ class Image extends React.Component {
     if (fluid) {
       const imageVariants = fluid
       const image = getCurrentSrcData(fluid)
+      const imageFileType = image.src.substring(
+        image.src.lastIndexOf(`.`) + 1,
+        image.src.length || image.src
+      )
+      const imageMediaType = imageMediaTypes[imageFileType]
```

Comment:
You're not enforcing lowercase here, so any `src` provided that has supported image extensions will not be assigned a `type` attribute value as it'd be `undefined`.

Probably best to centralize the logic too, instead of duplicating it for both fluid/fixed branches, co-locate the function beside the `imageMediaTypes` object instead? Then this can just be a single line.